File: content/showcase/12-macro/macro.lacking.rs
// lacking: prefix only; `4 twice` cannot be written
// a macro rewrites tokens, written in Rust and used
// in the same file; prefix only, never `4 twice`
macro_rules! twice {
    ($e:expr) => { $e * 2 };
}

fn main() {
    println!("{}", 3 + twice!(4)); // 11
}

File: content/showcase/12-macro/macro.lacking.c
// lacking: prefix only; `4 TWICE` cannot be written
// a macro rewrites text before the compiler reads it;
// prefix only, never `4 TWICE`
#include <stdio.h>
#define TWICE(e) ((e) * 2)

int main(void) {
    printf("%d\n", 3 + TWICE(4)); /* 11 */
    return 0;
}
